app/controllers/core_induction_programme/lessons_controller.rb

```diff
@@ -22,20 +22,24 @@ def show
   def edit; end
 
   def update
-    @course_lesson.save!
-    flash[:success] = "Your changes have been saved"
-    redirect_to cip_year_module_lesson_url
+    @course_lesson.assign_attributes(course_lesson_params)
+
+    if @course_lesson.update(course_lesson_params)
```

Comment:
Once we have `update` we don't need `assign_attributes` :)

`update(xxx)` = `assign_attibutes(xxx) + save`. 
`save` validates, adds errors to the object, and returns true or false based on whether validation passed. One of those magic conventions of Rails :)